packages/flutter/lib/src/painting/image_cache.dart

```diff
@@ -285,18 +286,17 @@ class ImageCache {
     }
   }
 
-  void _trackLiveImage(Object key, _CachedImage image) {
+  void _trackLiveImage(Object key, _LiveImage image, { bool putOk = true }) {
     // Avoid adding unnecessary callbacks to the completer.
     _liveImages.putIfAbsent(key, () {
+      assert(putOk);
       // Even if no callers to ImageProvider.resolve have listened to the stream,
       // the cache is listening to the stream and will remove itself once the
       // image completes to move it from pending to keepAlive.
       // Even if the cache size is 0, we still add this listener.
-      image.completer.addOnLastListenerRemovedCallback(() {
-        _liveImages.remove(key);
-      });
+      image.completer.addOnLastListenerRemovedCallback(image.remover);
       return image;
-    });
+    }).sizeBytes ??= image.sizeBytes;
```

Comment:
Why this addition? This doesn't seem to be covered by any tests, either. Can you add one?

Reply:
This now gets called where previously we just over-wrote the cache key with a new entry.  We can't overwrite the entry anymore because that would overwrite the thing that holds the remove callback.

I'll add a test.